### ISSUE TITLE
Zombie hordes flee north

### ISSUE THREAD

**Issue by Seraph062** [CleverRaven/Cataclysm-DDA#73725]
### Describe the bug

Hoards spawn in game and when player is near flee north. This makes makes them trivial to deal with.  

### Attach save file

N/A

### Steps to reproduce

1. Locate a zombie horde on overmap.
2. Approach horde
3. Watch horde flee north 

### Expected behavior

With "Wandering hordes" set to "False" I expect hordes to linger in an area until they agro onto the player (or another target).  

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19045.4291 (22H2)
- Game Version: 9e77385 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Slowdown Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions],
    Stats Through Skills [StatsThroughSkills],
    SpeedyDex [speedydex]
]

**Comment by RenechCDDA** [comment]
Please include the save for this.

**Comment by Seraph062** [comment]
I've started a new world on the latest 0.H release candidate. In this one all the hordes seem to flee west instead of north, but the effect is the same. If you move north from the location the game is saved at you will see the zombies all stream to the left. The only thing that seems to break this is the presence of civilians, which pull the zombies out of their behavior when the agro onto the civilian.   
[Test world.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/15327908/Test.world.zip)

**Comment by dead-letter-office** [comment]
I've noticed that when I first meet a horde, they all immediately start running together in a single direction. I don't find that this is consistently North, West, etc. It's often in the opposite direction that I approach them from.

By turning invisible and teleporting around I found that they seem to be locking on to the closest (distant) civilian from miles away and all rushing them. If I teleport to the center of a city, note the position of a civilian, and teleport next to them, from that perspective every zombie is running toward me.
